### ISSUE TITLE
pickFile with transform to get filename+ext but without full path

### ISSUE THREAD

**Issue by samuelbodin** [rioj7/command-variable#48]
Hi! I'm really glad that I found this plugin - thanks for putting in the time to make it! 

My question is related to the use of `pickFile` and how to use transformations. I want to be able to select a file without the path, for example `c:\foo\bar\file.mzn` should be transformed into `file.mzn`. The documentation was not super clear to me and I have experimented for a while, but without success. 

My current build task is defined as follows,
```
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "model + data",
      "type": "shell",
      "command": "minizinc -f ${input:model} instance-1.dzn",
      "problemMatcher": [],
      "group": {
        "kind": "build",
        "isDefault": true
      }
    }
  ],
  "inputs": [
    {
      "id": "model",
      "type": "command",
      "command": "extension.commandvariable.file.pickFile",
      "args": {
        "description": "Select a model",
        "display": "fileName",
        "include": "**/*.mzn",
        "transform": {
          "text": "${pickFile:model}",
          "find": ".*([a-zA-Z0-9-_]+\.mzn)$",
          "replace": "$1"
        }
      }
    }
  ]
}
```

The resulting command that is run in my shell is `minizinc -f c:\foo\bar\file.mzn instance-1.dzn` which unfortunately does not work for me - it needs to be `minizinc -f file.mzn instance-1.dzn`. It would be very appreciated if you could take a look and steer me in the right direction :) Thanks

**Comment by rioj7** [comment]
@samuelbodin `pickFile` does not yet support a `transform` property. I will consider this syntax for issue #18.

in issue #46 arquer shows a use of the [transform command to manipulate the picked path](https://github.com/rioj7/command-variable/issues/46#issuecomment-1374749753)

**Comment by samuelbodin** [comment]
Thanks @rioj7 for your answer! I followed the comment by arquer, but it's not playing nice when I enter my regex. I have the following configuration, which gives the error `Invalid regular expression: /^(.*)\([a-z_0-9]+mzn)$/: Unmatched ')'`. You can see that the `\\`that I have in `find` is to match on a backslash, but the final resulting regex, which is shown in the error, only seems to contain a single `\`. This of course causes it to escape the `(` character instead, hence the error. Any ideas on how I can get around this? Essentially all I want to do it is remove the path, which on windows is made up of backslashes. 

```
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "echo",
      "type": "shell",
      "command": "echo ${fileDirname}",
      "problemMatcher": [],
      "group": {
        "kind": "test",
        "isDefault": false
      }
    },
    {
      "label": "Minizinc",
      "type": "shell",
      "command": "minizinc -s --solver gecode -f ${input:model} data/to_jing_0.dzn",
      "problemMatcher": [],
      "group": {
        "kind": "build",
        "isDefault": true
      }
    }
  ],
  "inputs": [
    {
      "id": "model",
      "type": "command",
      "command": "extension.commandvariable.transform",
      "args": {
        "text": "${pickFile:mzn}",
        "find": "^(.*)\\([a-z_0-9]+\.mzn)$",
        "replace": "$1",
        "pickFile": {
          "mzn": {
            "description": "Select mzn",
            "fromFolder": {
              "fixed": "${workspaceFolder}/"
            },
            "include": "**/*.mzn",
            "showDirs": false,
            "display": "fileName"
          }
        }
      }
    }
  ]
}
```

**Comment by rioj7** [comment]
@samuelbodin  in #47 nabelekt shows how to do it:

```
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "Minizinc",
      "type": "shell",
      "command": "minizinc -f ${input:model} instance-1.dzn",
      "problemMatcher": [],
      "group": {
        "kind": "build",
        "isDefault": true
      }
    }
  ],
  "inputs": [
    {
      "id": "model",
      "type": "command",
      "command": "extension.commandvariable.transform",
      "args": {
        "text": "${pickFile:mzn}",
        "find": "^.*\\\\([a-z_0-9]+\\.mzn)$",
        "replace": "$1",
        "pickFile": {
          "mzn": {
            "description": "Select mzn",
            "fromFolder": {
              "fixed": "${workspaceFolder}/"
            },
            "include": "**/*.mzn",
            "display": "fileName"
          }
        }
      }
    }
  ]
}
```

With the v1.47.0 update you can shorten this to

```
  "inputs": [
    {
      "id": "model",
      "type": "command",
      "command": "extension.commandvariable.file.pickFile",
      "args": {
        "transform": { "text": "${fileBasename}" },
        "description": "Select mzn",
        "fromFolder": {
          "fixed": "${workspaceFolder}/"
        },
        "include": "**/*.mzn",
        "display": "fileName"
      }
    }
  ]
```

**Comment by samuelbodin** [comment]
Awesome stuff! `"find": "^.*\\\\([a-z_0-9]+\\.mzn)$"` worked perfectly. Much appreciated @rioj7!
